testutil/network/validator.go

```diff
@@ -0,0 +1,84 @@
+package network
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/cometbft/cometbft/node"
+	cmtclient "github.com/cometbft/cometbft/rpc/client"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/api"
+	srvconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Validator defines an in-process CometBFT validator node. Through this object,
+// a client can make RPC and API calls and interact with any client command
+// or handler.
+type Validator struct {
```

Comment:
If the goal is to limit the usage of the fields with getters, why those fields are still public?